src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3572,36 +3572,47 @@ void func_80836BEC(Player* this, PlayState* play) {
         ignoreLeash = true;
     }
 
-    cond = func_8083224C(play);
-    if (cond || (this->unk_66C != 0) || (this->stateFlags1 & (PLAYER_STATE1_12 | PLAYER_STATE1_25))) {
-        if (!cond) {
+    isTalking = func_8083224C(play);
+
+    if (isTalking || (this->unk_66C != 0) || (this->stateFlags1 & (PLAYER_STATE1_12 | PLAYER_STATE1_25))) {
+        if (!isTalking) {
             if (!(this->stateFlags1 & PLAYER_STATE1_25) &&
                 ((this->heldItemAction != PLAYER_IA_FISHING_POLE) || (this->unk_860 == 0)) &&
                 CHECK_BTN_ALL(sControlInput->press.button, BTN_Z)) {
-
                 if (this->actor.category == ACTORCAT_PLAYER) {
-                    actorToTarget = play->actorCtx.attention.naviHoverActor;
+                    // The next lock-on actor defaults to the actor Navi is hovering over.
+                    // This may change to the arrow hover actor below.
+                    nextLockOnActor = play->actorCtx.attention.naviHoverActor;
                 } else {
-                    actorToTarget = &GET_PLAYER(play)->actor;
+                    // Dark Link will always lock onto the player.
+                    nextLockOnActor = &GET_PLAYER(play)->actor;
                 }
 
-                holdTarget = (gSaveContext.zTargetSetting != 0) || (this->actor.category != ACTORCAT_PLAYER);
+                // Get saved Z Target setting.
+                // Dark Link uses Hold Targeting.
+                isHoldTargeting = (gSaveContext.zTargetSetting != 0) || (this->actor.category != ACTORCAT_PLAYER);
+
                 this->stateFlags1 |= PLAYER_STATE1_15;
 
-                if ((actorToTarget != NULL) && !(actorToTarget->flags & ACTOR_FLAG_LOCK_ON_DISABLED)) {
-                    if ((actorToTarget == this->focusActor) && (this->actor.category == ACTORCAT_PLAYER)) {
-                        actorToTarget = play->actorCtx.attention.arrowHoverActor;
+                if ((nextLockOnActor != NULL) && !(nextLockOnActor->flags & ACTOR_FLAG_LOCK_ON_DISABLED)) {
+                    // Navi hovers over the current lock-on actor, so `nextLockOnActor` and `focusActor`
+                    // will be the same if already locked on.
+                    // In this case, `nextLockOnActor` will be the arrow hover actor instead.
+                    if ((nextLockOnActor == this->focusActor) && (this->actor.category == ACTORCAT_PLAYER)) {
```

Comment:
which if does this comment apply to?

Reply:
below

Reply:
then I'd do either
```
if() {

  //
  if() {
    ...
```
or
```
if() {
  if() {
    //

    ...
```

(nitpicking I know 8) )

Reply:
I dont like the latter because of my explanation below (https://github.com/zeldaret/oot/pull/2172#discussion_r1749364081)
And I dont like the former either but ill do it if its a dealbreaker

Reply:
well in the current configuration I'm genuinely confused by which side of the comment the comment is talking about

Reply:
is it still confusing after reading the comment and realizing its focusing on `if (nextLockOnActor == this->focusActor)`? I get being confused when just scanning through it quickly

Reply:
as you described, you put the description for the if before and the description for what happens inside?

Then maybe the comment should be split like

```suggestion
                if ((nextLockOnActor != NULL) && !(nextLockOnActor->flags & ACTOR_FLAG_LOCK_ON_DISABLED)) {
                    // Navi hovers over the current lock-on actor, so `nextLockOnActor` and `focusActor`
                    // will be the same if already locked on.
                    if ((nextLockOnActor == this->focusActor) && (this->actor.category == ACTORCAT_PLAYER)) {
                        // In this case, `nextLockOnActor` will be the arrow hover actor instead.
```

?

Reply:
it awkwardly breaks up the flow imo, these make sense together. Id rather a line break over this if it fixes your concern